gui/webdisplay/src/RWebDisplayHandle.cxx

```diff
@@ -250,17 +250,12 @@ RWebDisplayHandle::BrowserCreator::Display(const RWebDisplayArgs &args)
 
    // these are secret parameters, hide them in temp file
    if ((url.find("token=") || url.find("key=")) && !args.IsBatchMode() && !args.IsHeadless()) {
-      gRandom->SetSeed(0);
-
-      tmpfile = gSystem->TempDirectory();
+      TString filebase = "root_start_";
 
-      if (tmpfile.back() != std::filesystem::path::preferred_separator)
-         tmpfile += std::filesystem::path::preferred_separator;
-      tmpfile += "root_start_"s + std::to_string(gRandom->Integer(0x100000)) + ".html";
+      auto f = gSystem->TempFileName(filebase, nullptr, ".html");
```

Comment:
One more comment: I think in the error cases before `RWebBrowserHandle` is returned, we will leak the temporary file.